plugins/chat/app/controllers/chat/emojis_controller.rb

```diff
@@ -3,7 +3,15 @@
 module Chat
   class EmojisController < ::Chat::BaseController
     def index
-      emojis = Emoji.all.group_by(&:group)
+      emoji_deny_list = SiteSetting.emoji_deny_list
+
+      if emoji_deny_list.present?
+        denied_emojis = emoji_deny_list.split("|")
+        emojis = Emoji.all.filter { |e| !denied_emojis.include?(e.name) }.group_by(&:group)
```

Comment:
I do wonder if we should introduce a `Emoji.allowed_emojis` method instead and make `Emoji.all` method private. Otherwise, any developer in the future that uses `Emoji.all` can potentially leak a denied emoji unless they know about the `emoji_deny_list` site setting.

Reply:
Also, I think we need to handle emoji aliases here as well.